pkg/kclient/services.go

```diff
@@ -0,0 +1,36 @@
+package kclient
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateService generates and creates the service
+// commonObjectMeta is the ObjectMeta for the service
+func (c *Client) CreateService(commonObjectMeta metav1.ObjectMeta, svcSpec corev1.ServiceSpec) (*corev1.Service, error) {
+	svc := corev1.Service{
+		ObjectMeta: commonObjectMeta,
+		Spec:       svcSpec,
+	}
+
+	service, err := c.KubeClient.CoreV1().Services(c.Namespace).Create(&svc)
```

Comment:
So I tried `odo push` with multiple components having the same endpoints, and i noticed that there is no devfile validation..

`Services.Create()` fails because it cannot create service with two endpoints, but this results in a deployment and a pod that is left hanging because it could not create pvc which is after service creation.

```
NAME                                    READY   STATUS    RESTARTS   AGE   LABELS
pod/spring-devfile-85d64fdd5b-dxn4j     0/2     Pending   0          9s    component=spring-devfile,pod-template-hash=4182098816

NAME                                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE   LABELS
deployment.extensions/spring-devfile     1         1         1            0           9s    component=spring-devfile
```

Maybe we should validate the data in devfile before we do `odo push` with the data but I think we dont have it now. Could be done later with an another PR perhaps, if we want to do it...

Reply:
Added validation for duplicate ports